### ISSUE TITLE
can't allocate memory: you tried to allocate 20484980736 bytes

### ISSUE THREAD

**Issue by suyuan945** [gafniguy/4D-Facial-Avatars#23]
![1646635349(1)](https://user-images.githubusercontent.com/26075641/156981977-3b199a2b-3cd6-4ed0-bcff-97d7733e5f1f.png)

This error occurred when I used the dataset you provided. I want to know what the hardware Env of the code are?
my pc: 
GPU:2080ti    16G
CPU memory：32G

**Comment by gafniguy** [comment]
This is most likely an Out of memory issue since the script tries to load all the training data to RAM, I think I was typically allocating around 70Gb.

You can bypass it by loading one sample at a time instead of loading everything at the start, in a fashion more similar to traditional DL pipelines on large datasets.

https://gist.github.com/gafniguy/5a66f471ad227d022aed96944432adea

**Comment by suyuan945** [comment]
Thank you for your reply. I'll try it.

**Comment by bd20222** [comment]
I found the data loading costs ~1 hr for my Linux machine. The available RAM is 80GB. I tried to reduce the number of images of the training and test set. Unfortunately, it still happens. Any ideas?

**Comment by gafniguy** [comment]
did you try to use the torch dataset class I attached here?
Instead of hogging everything on the RAM, it loads the images once at a time. It should be quite quick to start (just reads the jsons and not the images)

> I found the data loading costs ~1 hr for my Linux machine. The available RAM is 80GB. I tried to reduce the number of images of the training and test set. Unfortunately, it still happens. Any ideas?

**Comment by bd20222** [comment]
Will try it later. Thank you.
